cmd/kubeadm/app/constants/constants.go

```diff
@@ -208,8 +208,23 @@ var (
 
 	// MinimumKubeletVersion specifies the minimum version of kubelet which kubeadm supports
 	MinimumKubeletVersion = version.MustParseSemantic("v1.8.0")
+
+	// SupportedEtcdVersion lists officially supported etcd versions with corresponding kubernetes releases
+	SupportedEtcdVersion = map[uint8]string{
+		8: "3.0.17",
+		9: "3.1.10",
+	}
 )
 
+// GetEtcdSupportedVersion returns officially supported version of etcd for a specific kubernetes release
+// if passed version is not listed, the function returns empty string and an error
+func GetEtcdSupportedVersion(ver uint8) (string, error) {
```

Comment:
nit: please pass in the `*version.Version` object here and do `k8sVersion.Minor()` for the switching instead of forcing the consumer of this func to understand what uint var is needed

Reply:
I'd also like an unit test here